controllers/smbcommonconfig_controller.go

```diff
@@ -57,12 +58,28 @@ func (r *SmbCommonConfigReconciler) Reconcile(
 	// ---
 	log := r.Log.WithValues("smbcommonconfig", req.NamespacedName)
 
+	// Process OpenShift logic in one of two states:
+	// 1) Unknown cluster type due to first-time reconcile
+	// 2) Known to be running over OpenShift by in-memory cached state from
+	//    previous reconcile loop.
+	if r.ClusterType != "" && r.ClusterType != resources.ClusterTypeOpenshift {
+		return ctrl.Result{}, nil
+	}
+
 	mgr := resources.NewOpenShiftManager(r.Client, log, conf.Get())
 	res := mgr.Process(ctx, req.NamespacedName)
 	err := res.Err()
 	if res.Requeue() {
 		return ctrl.Result{Requeue: true}, err
 	}
+
+	// Cache in-memory cluster-type to avoid extra network round-trips in next
+	// reconcile phase.
+	if r.ClusterType == "" {
+		r.Log.Info("Keep in-memory", "ClusterType", mgr.ClusterType)
```

Comment:
If I was an admin reading this and not a coder I may be confused by this log line. Maybe something like "Saving discovered cluster type" or something along those lines:

Reply:
ACK.